### DISCUSSION TITLE
What if I wanted to Integrate CLAP as a Plugin Host plugin in Unreal Engine

### DISCUSSION THREAD

**Discussion by seraklab** [free-audio/clap#395] [Q&A]
Hello, I am I am not really a sound or signal specialist. But when I first heard about CLAP I thought that the redistributability aspect of the licence would allow to create cool tools for videogames. I always wanted to have real time vst control in games, but the VST licence would have made it difficult to redistribute the game with the sound plugins integrated . 
I may be mistaken but CLAP appears to solve this issue ... 
So now we need the ability for game engines to act as a plugin host ... My idea was to make CLAP Nodes in Meta sound, that would connect to the buses of Unreal's sound engine ... is there any material I should look at if I wanted to integrate Clap in Unreal as a host?
Any advice is welcome, where should I look first, if there is a useful tutorial I could look at, etc ... It would be super helpful!

**Comment by baconpaul** [comment]
There's a few open source liberally licensed hosts around

The best, most likely, is the clap-wrapper project which projects a clap into a VST3, AUv2, and standalone, hosted here in this repo.
clap-info, also here, is a simple bootstrap loader that dumps information about a clap and has tools for things like scanning.
Finally the clap-host is a full fledged host written in QT which shows how to write an app with CLAP

Each of them are here.

Sorry to be brief - off to an event for the day - but wanted to give you those ideas to get you started

**Reply by seraklab** [reply]
Thank you Paul!
I will take a closer look at those ... 
I think I'll try a test build of the QT clap host to try to understand how it is supposed in interact within a host.
Clap-info seems promising also.

**Reply by baconpaul** [reply]
The wrapper do taste most generic host since we project it into multiple situations. Definitely worth a look there too and the ihost there may just do what you need
